src/dmiopt.rs

```diff
@@ -373,52 +383,142 @@ impl Keyword {
                 .find_map(|system_info: SMBiosSystemInformation<'_>| system_info.family())
                 .ok_or(BiosParseError::SystemFamilyNotFound),
             Keyword::BaseboardManufacturer => data
-                .find_map(|baseboard_info: SMBiosBaseboardInformation<'_>| {
-                    baseboard_info.manufacturer()
+                .map(|baseboard_info: SMBiosBaseboardInformation<'_>| baseboard_info.manufacturer())

```

Comment:
Cool finding. I didn't have such output, and I guess you were able to see it from some of the input files that you have.

Reply:
The Azure VM's have an interesting BIOS with many records for processor sockets and memory slots, so they are particularly good for testing this scenario.